orte/mca/plm/slurm/plm_slurm_module.c

```diff
@@ -165,6 +168,44 @@ static int plm_slurm_launch_job(orte_job_t *jdata)
     return ORTE_SUCCESS;
 }
 
+/* Slurm 19 has removed support for --cpu_bind, and now only
+*  uses --cpu-bind. Use this function to test for
+*  supported srun parameters.
+*/
+int srun_check_parms(char *const argv[])
+{
+    int status;
+    int srun_pid;
+
+    srun_pid = fork();
+    if (0 == srun_pid) {
+        FILE* file = fopen("/dev/null", "w");
+        int fd = fileno(file);
+
+        dup2(fd, 1);
+        dup2(fd, 2);
+
+        execvp(argv[0], argv);
+
+        close(fd);
+        exit(errno);
+    }
+    else{
+        /* Wait for child process to end */
+        wait(&status);
+
+        /* Once child process, srun, has ended check exit code */
+        if (WIFEXITED(status)) {
+            if (255 == WEXITSTATUS(status)) {
+                return(0);
+            }
+        }
+    }
+    /* All other exit codes mean parms are good */
+    return(1);
```

Comment:
I think the logic isn't quite right here -- e.g., what if `srun` fails due to a signal?

It might be easier to check for the one condition that is good, and all others are failures.

Isn't the only good case where `srun` exits with exit status 0?